src/hera/workflows/cron_workflow.py

```diff
@@ -28,42 +23,11 @@
     Workflow as _ModelWorkflow,
 )
 from hera.workflows.protocol import TWorkflow
-from hera.workflows.workflow import Workflow, _WorkflowModelMapper
-
-
-class _CronWorkflowModelMapper(_WorkflowModelMapper):
-    @classmethod
-    def _get_model_class(cls) -> Type[BaseModel]:
-        return _ModelCronWorkflow
-
-    @classmethod
-    def build_model(
-        cls, hera_class: Type[ModelMapperMixin], hera_obj: ModelMapperMixin, model: TWorkflow
-    ) -> TWorkflow:
-        assert isinstance(hera_obj, ModelMapperMixin)
-
-        for attr, annotation in hera_class._get_all_annotations().items():
-            if get_origin(annotation) is Annotated and isinstance(
-                get_args(annotation)[1], ModelMapperMixin.ModelMapper
-            ):
-                mapper = get_args(annotation)[1]
-                if not isinstance(mapper, _CronWorkflowModelMapper) and mapper.model_path[0] == "spec":
-                    # Skip attributes mapped to spec by parent _WorkflowModelMapper
-                    continue
-
-                # Value comes from builder function if it exists on hera_obj, otherwise directly from the attr
-                value = (
-                    getattr(hera_obj, mapper.builder.__name__)()
-                    if mapper.builder is not None
-                    else getattr(hera_obj, attr)
-                )
-                if value is not None:
-                    _set_model_attr(model, mapper.model_path, value)
-
-        return model
+from hera.workflows.resource_base import ModelMapper, Self, _get_model_attr
+from hera.workflows.workflow import Workflow
 
 
-class CronWorkflow(Workflow):
+class CronWorkflow(Workflow, traverse_mro=False):
```

Comment:
CronWorfklow is kind of weird (because it doesn't define all its fields, and retargets a bunch of those fields into a sub-object).

I added a traverse_mro flag to handle the way its implemented now. I'd probably personally just duplicate the set of fields with the different model_paths. But this is how it was implemented before, so I left it as-is.